script_multiprocess.py

```diff
@@ -0,0 +1,75 @@
+"""
+You can ignore this file. It is just present to help run a simple program to
```

Comment:
The core functionality already has tests. 

The issue I have with adding multiprocessing tests is that it is difficult to keep track of state change. I don't think it makes sense to mock out the behavior. Also, the state changes are all time dependent (ie. open -> half-open). This will be a fragile test case if we are dependent on timings.